### PR TITLE
FEATURE NMH-133: Rename products in search [skip-test]

### PR DESCRIPTION
**Pull request by KshitizGIT**
Jira Ticket [NMH-133](https://johnsnowlabs.atlassian.net/browse/NMH-133)

### CODE REVIEW

File: docs/_plugins/search_index.rb [hunk 1]
```diff
@@ -21,10 +21,26 @@ def <= x
...
+      return old_product.sub(key, value)
+    end
+  end
+  old_product
+end
 def compatible_editions(editions, model_editions, edition)
   return edition if edition.to_s.empty?
 
-  outdated_editions = ['Spark NLP 2.1', 'Spark NLP for Healthcare 2.0']
+  outdated_editions = ['Spark NLP 2.1', 'Healthcare 2.0']
```
Comment (pabla) [inline docs/_plugins/search_index.rb:43]: Healthcare NLP 2.0

File: docs/_plugins/search_index.rb [hunk 2]
```diff
@@ -21,10 +21,26 @@ def <= x
...
 
+def edition_name(old_product)
+  old_to_new_product_name = {
+    'spark nlp for legal'=> 'Legal NLP',
+    'spark nlp for finance'=> 'Finance NLP',
+    'spark ocr'=> 'Visual NLP',
+    'spark nlp for healthcare'=> 'Healthcare NLP',
+    'spark nlp'=> 'Spark NLP',
+  }
+  old_product = old_product.downcase()
```
Comment (pabla) [inline docs/_plugins/search_index.rb:32]: Why do you need `downcase`?

File: docs/_frontend/src/models/FilterForm/FilterForm.js [hunk 3]
```diff
@@ -1,7 +1,14 @@
 import { createElement as e } from 'react';
 import Select from '../Select';
 import './FilterForm.css';
-import { products, productDisplayName } from '../ModelItem/utils';
```
Comment (pabla) [inline docs/_frontend/src/models/FilterForm/FilterForm.js:4]: Let's don't drop the mapping between old and new names. If we deploy without changes in the Elasticsearch index, the website will display old names.

**Review by pabla — COMMENTED** [review]
Remember, that we should support old and new product names on frontend and backend (search):
https://nlp.johnsnowlabs.com/models?edition=Healthcare+NLP+4.1
https://nlp.johnsnowlabs.com/models?edition=Spark+NLP+Healthcare+4.1

**Comment by KshitizGIT** [comment]
> Remember, that we should support old and new product names on frontend and backend (search): https://nlp.johnsnowlabs.com/models?edition=Healthcare+NLP+4.1 https://nlp.johnsnowlabs.com/models?edition=Spark+NLP+Healthcare+4.1

@pabla , Why do we need to support both if every models will use the new name after this PR ?

File: docs/_plugins/search_index.rb (hunk 2 above)
Comment (KshitizGIT) [reply docs/_plugins/search_index.rb]: Removed

File: docs/_frontend/src/models/FilterForm/FilterForm.js (hunk 3 above)
Comment (KshitizGIT) [reply docs/_frontend/src/models/FilterForm/FilterForm.js]: Done

**Comment by KshitizGIT** [comment]
> Remember, that we should support old and new product names on frontend and backend (search): https://nlp.johnsnowlabs.com/models?edition=Healthcare+NLP+4.1 https://nlp.johnsnowlabs.com/models?edition=Spark+NLP+Healthcare+4.1

@pabla , This PR allows supporting both variants on the search service

**Review by pabla — APPROVED** [review]

**Review by maziyarpanahi — APPROVED** [review]

**Pull request merged**